### DISCUSSION TITLE
Guest spawning inside park

### DISCUSSION THREAD

**Discussion by MaxMallon** [OpenRCT2/OpenRCT2#21100] [Feature requests]
I would like for an possibility for guests to spawn inside the park to be added.
Now, if you try to place a guest spawn point on a tile that is 'owned', it will complain. And if you make a tile that already has a spawn point owned, the spawn point will disappear. For normal scenarios etc, this is all fine.
But a lot of parks, especially contest entries, will not be a traditional park, but only part of a park, or something else entirely. We don't always want a park entrance. Now we have to fiddle with it, placing the entrance somewhere, hiding it away, making sure the path both works, and looks good, etc. It's a hassle.

My suggestion: allow spawn points on tiles that are part of the park, and guests spawning from there will simply 'already' be a guest. 
Guests wanting to leave the park will look for an park entrance, and if there are none, will instead navigate to the nearest spawn point itself, to be able to leave the map again.

**Comment by ZehMatt** [comment]
How would those guests leave the park?

**Reply by MaxMallon** [reply]
I.. literally explained that in my text..?
Last sentence: "Guests wanting to leave the park will look for a park entrance, and if there are none, will instead navigate to the nearest spawn point itself, to be able to leave the map again."

**Reply by ZehMatt** [reply]
So they just disappear when they reached the spawnpoint?

**Reply by ZehMatt** [reply]
Have to admit that part slipped.

**Reply by MaxMallon** [reply]
If they are in the 'leaving park' mode, and reach the spawnpoint, they'd just despawn yeah. 
Probably best to only have leaving-park mode guests do so, otherwise you're gonna have a lot of guests despawn as they randomly bump into the point.

**Comment by ocalhoun6** [comment]
Maybe this could be accomplished *relatively* simply by allowing spawn points to be freely moved with the tile inspector?

**Reply by ZehMatt** [reply]
That doesn't change the guests behavior, they would walk towards the park entrance even when the spawn would be inside the park. There is quite a bit of code to handle guests entering and leaving parks.

**Comment by Ghostbird** [comment]
I'm looking for something similar. I'm making an island park and I'd like to have the guests enter the map by "disembarking" from a ship onto a pier. Is that currently completely impossible, or could it be achieved with some save-game hacking?

> [!NOTE]
> I _do_ have an entrance.
